### PR TITLE
DEV-667: stage item in repo & index with catalog & full-text

### PR DESCRIPTION
**Pull request by aelkiss**
A couple questions that are coming up as I work through this:

* To what extent should individual things have their own docker-compose files, to what extent should they include dependencies and what should the default practices for port binding be? I think my inclination for the babel apps is to keep it fairly limited - enough to run any tests that might be present, but not replicating the full stack of dependencies for each app, and instead to keep that here.

* Where should the apache configuration go? My inclination would be to keep it here rather than in imgsrv, since it will serve for the other CGI applications (or should)

* Should we rename https://github.com/hathitrust/imgsrv-sample-data ? Does it make sense to keep it separate, or should we merge it with this repository?

* Things are somewhat inconsistent in terms of their reference to `mysql-sdr` vs `mariadb`. For consistency with current production and to minimize pain I think it makes sense to standardize on the existing `mysql-sdr` (and likewise for `solr-sdr-catalog`, perhaps?) but I could be convinced otherwise.

### CODE REVIEW

**Comment by aelkiss** [comment]
I think it does *not* make sense to containerize `stage-item`, largely because it itself runs things with docker-compose. I suspect just installing Ruby and the dependencies (they're fairly minimal here) on the Mac is easier than figuring out how to use the Docker API to start up stuff from within a Docker container (I'm sure it can be done, just seems like a more difficult yak to shave)

File: stage-item/Gemfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+source "https://rubygems.org"
+
+git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
+
+# gem "rails"
+
+gem "marc", "~> 1.2"
+gem "faraday", "~> 2.7"
+gem "faraday-follow_redirects"
+gem "ht-pairtree", git: "../../ht-pairtree"
```
Comment (aelkiss) [inline stage-item/Gemfile:12]: needs to be updated to point to https://github.com/hathitrust/ht-pairtree/ after that's merged

**Comment by aelkiss** [comment]
@moseshll I think for you the task (when this is all ready) will be to try to pull it down & run through the instructions (which are not up to date yet)

**Comment by aelkiss** [comment]
Regarding DEV-663 and pageturner: There was some support here for pageturner, which this stomps on. For DEV-663 I think it will make sense to re-add the apache support here, using the stuff done for imgsrv where possible, and then clean up some of the docker & apache stuff in imgsrv?

Out of scope for now, but should be addressed in DEV-663: the links in the catalog for items don't go to :8888 where the babel apps are running.

**Comment by aelkiss** [comment]
Building the imgsrv apache from here isn't working; the most expedient thing might just be to move that part in here now (since I was thinking of doing that anyway)

File: README.md [hunk 2]
```diff
@@ -48,24 +41,69 @@ docker-compose -f ./babel-local-dev/docker-compose.yml up
...
+* catalog: `http://localhost:8080/Search/Home`
+* catalog solr: `http://localhost:9033`
+* full-text solr: `http://localhost:8983`
+
+imgsrv:
 
+* `http://localhost:8888/cgi/imgsrv/cover?id=test.pd_open`
+* `http://localhost:8888/cgi/imgsrv/image?id=test.pd_open&seq=1`
+* `http://localhost:8888/cgi/imgsrv/html?id=test.pd_open&seq=1`
+* `http://localhost:8888/cgi/imgsrv/download/pdf?id=test.pd_open&seq=1&attachment=0`
```
Comment (respinos) [inline README.md:53]: My original vision was that you could proxy to the catalog fcgi app from so everything would run on the same port. If that's not a useful objective, we'll have to modify `common` to have known ports for catalog/babel.

Comment (aelkiss) [reply README.md:53]: We can probably do that...

Comment (aelkiss) [reply README.md:53]: This is working now.

**Comment by aelkiss** [comment]
Note that we should not merge this (even if working) until these PRs are merged:

https://github.com/hathitrust/ht-pairtree/pull/1
https://github.com/hathitrust/slip/pull/1
https://github.com/hathitrust/lss_solr_configs/pull/5
https://github.com/hathitrust/imgsrv/pull/11

**Comment by aelkiss** [comment]
I also see a complaint from pageturner that it can't read the `pod` table.

File: setup.sh [hunk 3]
```diff
@@ -8,6 +8,7 @@ git clone --recurse-submodules git@github.com:hathitrust/imgsrv
 git clone --recurse-submodules git@github.com:hathitrust/catalog
 git clone --recurse-submodules git@github.com:hathitrust/common
 git clone --recurse-submodules git@github.com:hathitrust/pt
+git clone --recurse-submodules git@github.com:hathitrust/ssd
```
Comment (moseshll) [inline setup.sh:12]: Running this script catapulted me into `gh auth` hell. I had hoped installing and using `gh` as my auth helper would be a more "modern" way of doing things -- compared to my previous SSH shenanigans and just checking everything out with my PAT. Alas, `gh` seems to really hate the mixing of Git protocols. I'll spend some more time with this when not crunched for time.

Comment (aelkiss) [reply setup.sh:12]: Hm, OK. Maybe the best practice would be to use https by default? Not sure.

Comment (aelkiss) [reply setup.sh]: @moseshll Now it asks if you want to use https or ssh before cloning.

**Comment by aelkiss** [comment]
As of right now the catalog links seem to magically be using http and not https... Not sure what's going on

**Comment by aelkiss** [comment]
@moseshll @respinos I was able to run through the instructions in the README, and I believe everything now works both over ssh and https, and without any permissions weirdness. I think we still need to get these PRs merged:

https://github.com/hathitrust/lss_solr_configs/pull/5
https://github.com/hathitrust/pt/pull/4
https://github.com/hathitrust/ssd/pull/1

Once those are merged I will remove the branches from `setup.sh` here, squash the commits here, and then I think we should be ready to merge this!

File: setup.sh (hunk 3 above)
Comment (aelkiss) [reply setup.sh]: And I've updated the submodules to use relative paths, so either will work.

**Comment by aelkiss** [comment]
> As of right now the catalog links seem to magically be using http and not https... Not sure what's going on

This seems to be some issue with my browser - if I try in a new private/incognito window it's fine. Must be some sort of pinning to localhost to https :weary:

**Comment by aelkiss** [comment]
Still unresolved; should track; probably don't need to solve right this minute (but should soon, and will need to for mb, etc; I'll make a note in that issue)

* use database user that has permission to ht sessions table

Also @moseshll @carylwyatt when you have a chance it would be good to make sure this all works on ARM without emulation/rigmarole - I think it should but I don't know for sure.

**Comment by aelkiss** [comment]
Now just waiting on https://github.com/hathitrust/lss_solr_configs/pull/5

**Comment by aelkiss** [comment]
Superceded by #5

**Pull request closed without merging**